### ISSUE TITLE
Security problem / sharing options

### ISSUE THREAD

**Issue by MathiasPhilipp** [nextcloud/server#8028]
I'm not a developer and absolutely new here.  I'm admin and user, never worked with github. Sorry if I do anything wrong or at the wrong place.

In my opinion, the following seems to be a security problem:

Combining the options
"Restrict users to share only with users in their group"
and
"Allow username autocompletion in share dialog." If this is disabled
the full username or email address needs to be entered "

If I enter the username with autocomplete when sharing, all hits for all users over all groups are displayed. Also the users of groups with whom I may not allowed to share. With it I can find out all users (with email) about all groups.

I would expect autocomplete to show me only the users I'm allowed to share.

Is this a security bug or how can I prevent it?

 
### Steps to reproduce
1. enable "Restrict users to share only with users in their group"
2. enable "Allow username autocompletion in share dialog..."
3. share

### Expected behaviour
I would expect autocomplete to show me only the users I'm allowed to share.

### Actual behaviour
all hits for all users over all groups are displayed. Also the users of groups with whom I may not allowed to share.

### Server configuration

**Operating system**:   ? web-hosting

**Web server:** ?? web-hosting

**Database:** MYSQL 5.7.20

**PHP version:**  7.2.1

**Nextcloud version:** 12.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:** Updeted from Nextcloud

**Where did you install Nextcloud from:** web-hosting

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by tflidd** [comment]
It was already part of this request (if I don't misread it: https://github.com/nextcloud/server/issues/5107) which was set to the Nextcloud 13 milestone. The discussion was more related about the contacts menu.

this point in #5107 seems still to be an issue:

>  If sharing is restricted to users own groups, he should only see contacts from his groups:

@LEDfan did your implementation (PR #5585) only consider the contacts menu but not the sharing menu?

I can reproduce this problem in NC 13 beta 3.

**Comment by tflidd** [comment]
is this the same?
https://github.com/nextcloud/server/issues/6530

**Comment by MathiasPhilipp** [comment]
I am not sure: 

In issue # 8028 you can avoid the problem by disabling "autocomplition"

 <https://github.com/nextcloud/server/issues/6530> #6530 seems to have nothing to do with "autocomplition".

 

I updated today from 12.0.4. to 12.0.5. 

#8028 seems to be fixed.

 

I’ll check systematically

 

 

Mit freundlichen Grüßen

Mathias Philipp

 

Von: tflidd [mailto:notifications@github.com] 
Gesendet: Samstag, 27. Januar 2018 22:33
An: nextcloud/server
Cc: MathiasPhilipp; Author
Betreff: Re: [nextcloud/server] Security problem / sharing options (#8028)

 

is this the same?
 <https://github.com/nextcloud/server/issues/6530> #6530

**Comment by MathiasPhilipp** [comment]
Yes 5107 corresponds to #8028.

 

After update 12.0.4. to 12.0.5. the sharing problem is already solved.

I can’t reproduce the security problem. From my point of view it’s fixed.

 

Mit freundlichen Grüßen

Mathias Philipp

 

Von: tflidd [mailto:notifications@github.com] 
Gesendet: Samstag, 27. Januar 2018 22:26
An: nextcloud/server
Cc: MathiasPhilipp; Author
Betreff: Re: [nextcloud/server] Security problem / sharing options (#8028)

 

It was already part of this request (if I don't misread it: #5107 <https://github.com/nextcloud/server/issues/5107> ) which was set to the Nextcloud 13 milestone. The discussion was more related about the contacts menu.

this point in #5107 <https://github.com/nextcloud/server/issues/5107>  seems still to be an issue:

If sharing is restricted to users own groups, he should only see contacts from his groups:

@LEDfan <https://github.com/ledfan>  did your implementation (PR #5585 <https://github.com/nextcloud/server/pull/5585> ) only consider the contacts menu but not the sharing menu?

I can reproduce this problem in NC 13 beta 3.

**Comment by tflidd** [comment]
> After update 12.0.4. to 12.0.5. the sharing problem is already solved. I can’t reproduce the security problem. From my point of view it’s fixed.

Except that it is broken for NC 13 RC 3 again....

**Comment by LEDfan** [comment]
> After update 12.0.4. to 12.0.5. the sharing problem is already solved.

This bug is the same as in #7428 and #6530 which is fixed in 12.0.5.

> I can reproduce this problem in NC 13 beta 3.

The fix was merged into beta 4, I just tested on stable13 and there it's still fixed.
